### PR TITLE
[Merged by Bors] - feat: Support podOverrides

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/346

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/zk_controller.rs [hunk 1]
```diff
@@ -88,6 +89,10 @@ pub enum Error {
         source: strum::ParseError,
         role: String,
     },
+    #[snafu(display("internal operator failure: {source}"))]
```
Comment (maltesander) [inline rust/operator-binary/src/zk_controller.rs:92]: No need to mention the source in the message.
```suggestion
    #[snafu(display("internal operator failure"))]
```

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -366,14 +374,34 @@ impl Configuration for ZookeeperConfigFragment {
...
+impl ZookeeperRole {
+    pub fn roles() -> Vec<String> {
+        let mut roles = vec![];
+        for role in Self::iter() {
+            roles.push(role.to_string())
+        }
+        roles
+    }
+}
+
```
Comment (maltesander) [inline rust/crd/src/lib.rs:404]: Do we really need this for one role?

**Review by maltesander — CHANGES_REQUESTED** [review]
Minor things.
Do we need an integration test? Currently if anything in the DeepMerge changes we wont fail anywhere?

File: rust/crd/src/lib.rs (hunk 2 above)
Comment (sbernauer) [reply rust/crd/src/lib.rs:404]: I just copy-pasted it from trino-op. We could hardcode the list, but I personally prefer it this way for
1. Consistency
2. If (given unlikely) we add a role we are not introducing a bug when we forget about this

**Comment by sbernauer** [comment]
I think the effort of maintaining this tests does not justify testing something coming from k8s-openapi.
The only thing we could test is our two `.merge()` statements, where it personally think it's not worth it.

File: rust/crd/src/lib.rs (hunk 2 above)
Comment (maltesander) [reply rust/crd/src/lib.rs:404]: Ok works for me.

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request closed without merging**